admin/module/catalog-admin-module-sources/src/main/java/org/codice/ddf/catalog/admin/plugin/AdminSourcePollerServiceBean.java

```diff
@@ -0,0 +1,242 @@
+
+    private static final String MAP_ENTRY_ENABLED = "enabled";
+
+    private static final String MAP_ENTRY_FPID = "fpid";
+
+    private static final String MAP_ENTRY_NAME = "name";
+
+    private static final String MAP_ENTRY_BUNDLE_NAME = "bundle_name";
+
+    private static final String MAP_ENTRY_BUNDLE_LOCATION = "bundle_location";
+
+    private static final String MAP_ENTRY_BUNDLE = "bundle";
+
+    private static final String MAP_ENTRY_PROPERTIES = "properties";
+
+    private static final String MAP_ENTRY_CONFIGURATIONS = "configurations";
+
+    private static final String DISABLED = "_disabled";
+
+    private static final String SERVICE_NAME = ":service=admin-source-poller-service";
+
+    private final ObjectName objectName;
+
+    private final MBeanServer mBeanServer;
+
+    private final BundleContext bundleContext;
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    public AdminSourcePollerServiceBean(ConfigurationAdmin configurationAdmin) {
+        bundleContext = getBundleContext();
+        this.configurationAdmin = configurationAdmin;
+        mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        ObjectName objName = null;
+        try {
+            objName = new ObjectName(AdminSourcePollerServiceBean.class.getName() + SERVICE_NAME);
+        } catch (MalformedObjectNameException e) {
+            LOGGER.error("Unable to create Admin Source Poller Service MBean with name [{}].",
+                    AdminSourcePollerServiceBean.class.getName() + SERVICE_NAME, e);
+        }
+        objectName = objName;
+    }
+
+    public void init() {
+        try {
+            try {
+                mBeanServer.registerMBean(this, objectName);
+                LOGGER.info(
+                        "Registered Admin Source Poller Service Service MBean under object name: {}",
+                        objectName.toString());
+            } catch (InstanceAlreadyExistsException e) {
+                // Try to remove and re-register
+                mBeanServer.unregisterMBean(objectName);
+                mBeanServer.registerMBean(this, objectName);
+                LOGGER.info("Re-registered Admin Source Poller Service Service MBean");
+            }
+        } catch (Exception e) {
+            LOGGER.error("Could not register MBean [{}].", objectName.toString(), e);
+        }
+    }
+
+    public void destroy() {
+        try {
+            if (objectName != null && mBeanServer != null) {
+                mBeanServer.unregisterMBean(objectName);
+                LOGGER.info("Unregistered Admin Source Poller Service Service MBean");
+            }
+        } catch (Exception e) {
+            LOGGER.error("Exception unregistering MBean [{}].", objectName.toString(), e);
+        }
+    }
+
+    @Override
+    public boolean sourceStatus(String servicePID) {
+        try {
+            List<ServiceReference<? extends Source>> sourceReferences = getServiceReferences();
```

Comment:
If you add a LDAP filter as a parameter, you can get a list filtered by service.pid.  That will remove a lot of this code.